src/lando/api/views.py

```diff
@@ -1 +1,111 @@
-# Create your views here.
+from functools import wraps
+from typing import Callable
+
+from django import forms
+from django.http import JsonResponse
+from django.views import View
+
+from lando.main.models.revision import DiffWarning, DiffWarningStatus
+from lando.utils.phabricator import get_phabricator_client
+
+
+def phabricator_api_key_required(func: callable) -> Callable:
+    """A simple wrapper that checks for a valid Phabricator API token."""
+
+    @wraps(func)
+    def _wrapper(self, request, *args, **kwargs):
+        HEADER = "X-Phabricator-Api-Key"
+        if HEADER not in request.headers:
+            return JsonResponse({"error": f"{HEADER} missing."}, status=400)
+
+        api_key = request.headers[HEADER]
+        client = get_phabricator_client(api_key=api_key)
+        has_valid_token = client.verify_api_token()
+
+        if not has_valid_token:
+            return JsonResponse({}, 401)
+
+        return func(self, request, *args, **kwargs)
+
+    return _wrapper
+
+
+class LegacyDiffWarningView(View):
+    """
+    This class provides the API controllers for the legacy `DiffWarning` model.
+
+    These API endpoints can be used by clients (such as Code Review bot) to
+    get, create, or archive warnings.
+    """
+
+    @classmethod
+    def _get_form(cls, method: str, data: dict):
```

Comment:
Nit: this function is harder to follow than if we directly build the form in the corresponding handler.

Reply:
Moved forms so they are in separate classes and in each handler.